src/components/visualizations/utils/color.ts

```diff
@@ -300,13 +312,18 @@ function getRGBColorCode(color: string | IColor): IColor {
     return color;
 }
 
-export function getColorPalette(baseColor: string | IColor): string[] {
+export function getColorPalette(baseColor: string | IColor, opacity: number = 1): string[] {
     const colorItemsCount = 6;
     const { r, g, b } = getRGBColorCode(baseColor);
     const channels = [r, g, b];
     const steps = channels.map(channel => (255 - channel) / colorItemsCount);
-    const generatedColors = getCalculatedColors(colorItemsCount, channels, steps);
-    return [...generatedColors.reverse(), formatColor(r, g, b)];
+    const generatedColors = getCalculatedColors(colorItemsCount, channels, steps, opacity);
+    return [...generatedColors.reverse(), formatColor(r, g, b, opacity)];
+}
+
+export function rgbToRgba(color: string, opacity: number = 1) {
```

Comment:
please add the return type for this function :)

Reply:
Updated